src/iuse_actor.cpp

```diff
@@ -231,77 +231,77 @@ std::optional<int> iuse_transform::use( Character &p, item &it, bool t, const tr
 
     int result = 0;
 
-    const bool possess = p.has_item( it ) ||
-                         ( it.has_flag( flag_ALLOWS_REMOTE_USE ) && square_dist( p.pos(), pos ) <= 1 );
+    const bool possess = p->has_item( it ) ||
```

Comment:
I would prefer a solution with less churn, something like
```c++
std::optional<int> iuse_transform::use( Character *c, item &it, bool t, const tripoint &pos ) const
    if( c == nullptr ) {
        debugmsg( "yo dawg that's not right" );
        return std::nullopt;
    }
    Character &p = *c;
```
then you won't have to change every instance of `p`. You can drop the nullptr check for all current functions too.

Reply:
I don't see what you are doing here. Changing `iuse_transform::use( Character &c` to `iuse_transform::use( Character *c` is all of the churn and without that churn the check you are suggesting is not possible.

Any kind of null check I tried to do with `Character &` resulted in `error: reference cannot be bound to dereferenced null pointer in well-defined C++ code; pointer may be assumed to always convert to true [clang-diagnostic-undefined-bool-conversion,-warnings-as-errors]` from our code analyzers that run on Github.